datatypes/src/primitives/bounding_box.rs

```diff
@@ -343,8 +343,17 @@ impl BoundingBox2D {
         self.upper_right_coordinate = self.upper_right_coordinate.max_elements(coord);
     }
 
+    pub fn extend_self(&mut self, other: &Self) {
+        self.lower_left_coordinate = self
+            .lower_left_coordinate
+            .min_elements(other.lower_left_coordinate);
+        self.upper_right_coordinate = self
+            .upper_right_coordinate
+            .max_elements(other.upper_right_coordinate);
+    }
+
     #[must_use]
-    pub fn extend(&mut self, other: &Self) -> Self {
+    pub fn union(&self, other: &Self) -> Self {
```

Comment:
is union really a better word here? If two bboxes are completely disjoint, is it still a union if we create a new one that contains both?